podcast/2.18.md

```diff
@@ -0,0 +1,545 @@
+
+**Beyang Liu:**:
+
+Yeah. Your experience resonates just so much from the kind of initial skepticism. Like we hear all the time. I almost divide the world into two camps of developers: those who get code search, and those who've never tried it before. And the ones who haven't tried it really just have a hard time comprehending how this is better than grep, or better than control-F in the editor. 
+
+And it is one of those things where, once you start using it, you start using it more and more. It's awesome to have all the code searchable in one place. And then the link sharing too, it just becomes almost this ... Like it starts out as a single-player tool because you're just searching for stuff for yourself. But then you start sharing links with colleagues. They can jump to what you're looking into and continue the exploration from there. It's kind of interesting that it's not more, I guess, universal at this point. I would say like Etsy and MailChimp are kind of the exception. Most engineering organizations don't have a code search tool, which is which interesting, no?
+
+**Kelly Norton:**
+
+Yeah, I agree. And one of the things that I really love is the subtle little interactions that people develop with a code search tool. So, as an example, MailChimp is a PHP code base. We do have some static analysis through a tool called Fan that was created by some other folks at Etsy, and is now a part of the PHP ecosystem. But deleting code can actually still be pretty sketchy when you're in a PHP code base. I mean, there are some weird ways to call methods in PHP.
+
+So there is a practice where if you are deleting a particularly important method in the MailChimp code base, and you feel like you've gotten all the references, you actually prove it to people by including the Hound, the basically ‘knows nothing found’ Hound link in your poll request. So we have a template, and one of them is you doing a quick risk assessment of your change. And people deleting code will often just post the Hound link as one of the risk analysis things they've done to show that the code is safe and that they've considered it. So I love little interactions like that because like those are the types of subtle interactions you can't really anticipate in advance.
```

Comment:
```suggestion
So there is a practice where if you are deleting a particularly important method in the MailChimp code base, and you feel like you've gotten all the references, you actually prove it to people by including the Hound, the basically ‘knows nothing found’ Hound link in your pull request. So we have a template, and one of them is you doing a quick risk assessment of your change. And people deleting code will often just post the Hound link as one of the risk analysis things they've done to show that the code is safe and that they've considered it. So I love little interactions like that because like those are the types of subtle interactions you can't really anticipate in advance.
```